### PR TITLE
Imx6 smc

### PR DESCRIPTION
**Pull request by natestuder**
Fixes to some small issues found when using the monitor mode hook on the iMX6.

### CODE REVIEW

File: elfloader-tool/src/plat/tk1/monitor.S [hunk 1]
```diff
@@ -64,10 +64,10 @@ smc_handler:
     ldr sp, [pc, #-12]
     push {lr}
     blx  r0
-    mrc p15, 0, r0, c1, c1, 0
+    mrc p15, 0, r3, c1, c1, 0
```
Comment (yyshen) [inline elfloader-tool/src/plat/tk1/monitor.S]: If we stick to the calling convention, r3 is also a return result register. Maybe we could use r12 instead?

File: elfloader-tool/src/plat/imx6/monitor.S [hunk 2]
```diff
@@ -65,10 +65,10 @@ smc_handler:
     push {lr}
     blx r0
     isb
-    mrc p15, 0, r5, c1, c1, 0
+    mrc p15, 0, r3, c1, c1, 0
     /* set the NS bit */
-    orr r5, r5, #1
-    mcr p15, 0, r5, c1, c1, 0
+    orr r3, r3, #1
```
Comment (yyshen) [inline elfloader-tool/src/plat/imx6/monitor.S]: same. could we use r12 instead of r3?

**Comment by yyshen** [comment]
Hi Nathan,

Thanks for the PR. Sorry for the arbitrary choice of registers and any painful debugging it may cause. The intention of the hook is to execute some a short sequence of instructions provided by the caller, so the caller should know which register should be saved. However, sticking to the calling convention would be helpful. I made some comments inline. Please let me know what you think.

File: elfloader-tool/src/plat/tk1/monitor.S (hunk 1 above)
Comment (natestuder) [reply elfloader-tool/src/plat/tk1/monitor.S]: That should work.  I'll send an updated commit.

**Comment by pingerino** [comment]
hey @natestuder, I'm happy to merge this, looks like it got lost during the holidays. Do you mind if I rebase it to avoid a merge commit? Otherwise, can you please rebase and merge?

**Comment by natestuder** [comment]
Rebased and tested.

**Comment by kent-mcleod** [comment]
Hey @natestuder, the original commits were accidentally merged, (https://github.com/seL4/seL4_tools/commit/f5fd7dab0d50e84cb8a14f661faeda9d39e74c33).  Are you happy with that or do you want us to revert the commits and apply the ones in this PR?

**Comment by natestuder** [comment]
Works for me.

**Pull request closed without merging**